<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>620</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label_image">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>500</width>
      <height>500</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Panel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <property name="text">
     <string>select a picture</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QStackedWidget" name="stackedWidget">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>150</y>
      <width>241</width>
      <height>381</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <stylestrategy>PreferDefault</stylestrategy>
      <kerning>true</kerning>
     </font>
    </property>
    <property name="cursor">
     <cursorShape>ArrowCursor</cursorShape>
    </property>
    <property name="focusPolicy">
     <enum>Qt::NoFocus</enum>
    </property>
    <property name="acceptDrops">
     <bool>false</bool>
    </property>
    <property name="toolTipDuration">
     <number>0</number>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="inputMethodHints">
     <set>Qt::ImhHiddenText</set>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Sunken</enum>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="region_grow_3">
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>40</y>
        <width>60</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>std decay</string>
      </property>
     </widget>
     <widget class="QPushButton" name="region_grow">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>150</y>
        <width>113</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>region grow</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="std_decay">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>60</y>
        <width>113</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>400</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="level_set_2">
     <widget class="QWidget" name="verticalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>211</width>
        <height>331</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string> timestep:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="time_step">
           <property name="text">
            <string>0.1</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>    epsilon:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="epsilon">
           <property name="text">
            <string>2.0</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>           mu:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="mu">
           <property name="text">
            <string>1.0</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>  lambda1:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lambd1">
           <property name="text">
            <string>1.0</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>      sigma:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="sigma">
           <property name="text">
            <string>3.0</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>       steep:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="c0">
           <property name="inputMethodHints">
            <set>Qt::ImhHiddenText</set>
           </property>
           <property name="text">
            <string>2.0</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="leftMargin">
          <number>2</number>
         </property>
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>  iter num:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="iter_num">
           <property name="text">
            <string>500</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_9">
         <item>
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string> lambda2:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lambda2">
           <property name="text">
            <string>1.0</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>           nu:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="nu">
           <property name="text">
            <string>0.003</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QPushButton" name="level_set">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>340</y>
        <width>113</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>level set</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="rectify">
     <widget class="QPushButton" name="gen_polygon">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>80</y>
        <width>113</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>gen shape</string>
      </property>
     </widget>
     <widget class="QPushButton" name="delete_polygon">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>190</y>
        <width>113</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>delete</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>100</y>
      <width>184</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_10">
     <item>
      <widget class="QLabel" name="label_11">
       <property name="text">
        <string>method:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="comboBox">
       <item>
        <property name="text">
         <string>region grow</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>level set</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>rectify</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLineEdit" name="file_path">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>60</y>
      <width>241</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>30</y>
      <width>60</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>filepath：</string>
    </property>
   </widget>
   <widget class="QPushButton" name="gen_file">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>20</y>
      <width>113</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>gen file</string>
    </property>
   </widget>
   <zorder>stackedWidget</zorder>
   <zorder>label_image</zorder>
   <zorder>horizontalLayoutWidget_2</zorder>
   <zorder>file_path</zorder>
   <zorder>label_12</zorder>
   <zorder>gen_file</zorder>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menufile">
    <property name="font">
     <font>
      <family>.SF NS Display</family>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="title">
     <string>file</string>
    </property>
    <addaction name="actionopen"/>
    <addaction name="separator"/>
    <addaction name="actionreset"/>
   </widget>
   <addaction name="menufile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionopen"/>
   <addaction name="actionreset"/>
  </widget>
  <action name="actionopen">
   <property name="text">
    <string>open</string>
   </property>
  </action>
  <action name="actionreset">
   <property name="text">
    <string>reset</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>level_set</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>exec_level_set()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>634</x>
     <y>548</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>543</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>change_method()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>630</x>
     <y>157</y>
    </hint>
    <hint type="destinationlabel">
     <x>652</x>
     <y>119</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>region_grow</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>exec_region_grow()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>564</x>
     <y>361</y>
    </hint>
    <hint type="destinationlabel">
     <x>524</x>
     <y>365</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>delete_polygon</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>exec_delete_polygon()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>578</x>
     <y>405</y>
    </hint>
    <hint type="destinationlabel">
     <x>526</x>
     <y>412</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gen_polygon</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>exec_gen_polygon()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>584</x>
     <y>287</y>
    </hint>
    <hint type="destinationlabel">
     <x>523</x>
     <y>286</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionopen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>open_file()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>401</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionreset</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>reset_image()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>401</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>exec_level_set()</slot>
  <slot>change_method()</slot>
  <slot>exec_region_grow()</slot>
  <slot>exec_delete_polygon()</slot>
  <slot>exec_gen_polygon()</slot>
  <slot>open_file()</slot>
  <slot>reset_image()</slot>
 </slots>
</ui>
